assets/src/domain/eventEditor/services/filterState/datetimes/useDatesListFilterStateManager.ts

```diff
@@ -24,45 +24,39 @@ const useDatesListFilterStateManager = (): FSM => {
 		}
 	};
 
-	const setDisplayStartOrEndDate: FSM['setDisplayStartOrEndDate'] = useCallback(
-		(displayStartOrEndDate) => {
-			dispatch({
-				type: 'SET_DISPLAY_START_OR_END_DATE',
-				displayStartOrEndDate,
-			});
-		},
-		[dispatch]
+	const setDisplayStartOrEndDate: FSM['setDisplayStartOrEndDate'] = useCallback((displayStartOrEndDate) => {
+		dispatch({
+			type: 'SET_DISPLAY_START_OR_END_DATE',
+			displayStartOrEndDate,
+		});
+	}, []);
+
+	const setSales: FSM['setSales'] = useCallback((sales) => {
+		resetPageNumber(sales);
+		dispatch({
+			type: 'SET_SALES',
+			sales,
+		});
+	}, []);
+
+	const setStatus: FSM['setStatus'] = useCallback((status) => {
+		resetPageNumber(status);
+		dispatch({
+			type: 'SET_STATUS',
+			status,
+		});
+	}, []);
+
+	return useMemo(
+		() => ({
+			...state,
+			...entityFilterState,
+			setDisplayStartOrEndDate,
+			setSales,
+			setStatus,
+		}),
+		[...Object.values(state), ...Object.values(entityFilterState.getState())]
```

Comment:
seems like there is too much logic happening here;
maybe `entityFilterState` can be memorized separately and by doing so, we wouldn't care about that here